### ISSUE TITLE
TS complain "Variable is used before being assigned." even assigned Variable in loop without any condition.

### ISSUE THREAD

**Issue by nurmohammed840** [microsoft/TypeScript#39572]
![image](https://raw.githubusercontent.com/nurmohammed840/ScreenShots/master/app.png)

TypeScript version : 3.9.6

I don't know this is a bug or not.
TypeScript complain "Variable 'num' is used before being assigned" even in `foreach` loop there is no condition to set `num = 0`
</br>So 100% sure that num would be `0` . Is not it?

**Comment by MartinJohns** [comment]
Duplicate of #9998. Would be solved by #11498.

**Comment by nurmohammed840** [comment]
```
// typescript version 3.9.2
interface Obj {
  a: number;
  b: number;
  c: number;
}

let obj2: Obj = {
  a: 2,
  b: 3,
  c: obj2.a + obj2.b,   // Block-scoped variable 'obj2' used before its declaration
};

let obj: Obj = {    // Property 'c' is missing in type '{ a: number; b: number; }' but required in type 'Obj'
  a: 2,
  b: 3,
};
obj.c = obj.a + obj.b; 
```
This code is 100% valid in JavaScript.
